compactor_scheduler/tests/local_scheduler/get_jobs.rs

```diff
@@ -0,0 +1,122 @@
+use std::sync::Arc;
+
+use assert_matches::assert_matches;
+use compactor_scheduler::{create_test_scheduler, CompactionJob};
+use data_types::PartitionId;
+use iox_tests::TestCatalog;
+
+use super::{super::helpers, create_scheduler_with_partition};
+
+#[tokio::test]
+async fn test_get_jobs_mocked_partition_ids() {
+    let catalog = TestCatalog::new();
+    let partitions = vec![PartitionId::new(0), PartitionId::new(1234242)];
+
+    let scheduler = create_test_scheduler(
+        catalog.catalog(),
+        Arc::clone(&catalog.time_provider()),
+        Some(partitions.clone()),
+    );
+
+    let mut result = scheduler
+        .get_jobs()
+        .await
+        .iter()
+        .map(|j| j.partition_id)
+        .collect::<Vec<PartitionId>>();
+    result.sort();
+
+    assert_eq!(result, partitions);
+}
+
+#[tokio::test]
+async fn test_get_jobs_with_utils_methods() {
+    test_helpers::maybe_start_logging();
+
+    let (scheduler, existing_1, existing_2, _) = create_scheduler_with_partition().await;
+    let jobs = scheduler.get_jobs().await;
+
+    assert_matches!(
+        jobs[..],
+        [CompactionJob { partition_id, .. }] if partition_id == existing_1.partition.partition.id && partition_id == existing_2.partition.partition.id,
+        "expect get_jobs() to return a single job, with two affiliated parquet files"
+    );
+}
+
+#[tokio::test]
+async fn test_get_jobs_will_not_fetch_leased_partitions() {
+    test_helpers::maybe_start_logging();
+
+    let (scheduler, existing_1, existing_2, _) = create_scheduler_with_partition().await;
+    let jobs = scheduler.get_jobs().await;
+    let existing_1_partition = existing_1.partition.partition.id;
+    let existing_2_partition = existing_2.partition.partition.id;
+
+    assert_matches!(
+        jobs[..],
+        [CompactionJob { partition_id, .. }] if partition_id == existing_1_partition && partition_id == existing_2_partition,
+        "has partition returned by get_jobs()"
+    );
+
+    assert_matches!(
+        scheduler.get_jobs().await[..],
+        [],
+        "no partitions found in next get_jobs() call"
+    );
```

Comment:
Yeah this would be a lot clearer as

```suggestion
    let jobs = scheduler.get_jobs().await;
    assert!(jobs.is_empty(), "Expected empty, got {:?}, jobs);
```

Reply:
The expected terms were cleaned up, across the integration test suite. TY for highlighting this need!